01.fit_a_line/README.md

```diff
@@ -116,49 +117,58 @@ When training complex models, we usually have one more split: the validation set
 
 `fit_a_line/trainer.py` demonstrates the training using [PaddlePaddle](http://paddlepaddle.org).
 
-### Initialize PaddlePaddle
+### Datafeeder Configuration
 
-```python
-paddle.init(use_gpu=False, trainer_count=1)
-```
+We first define data feeders for test and train. The feeder reads a `BATCH_SIZE` of data each time and feed them to the training/testing process. Users can shuffle a batch out of a `buf_size` in order to make the data random.
 
-### Model Configuration
+```python
+BATCH_SIZE = 20
 
-Linear regression is essentially a fully-connected layer with linear activation:
+train_reader = paddle.batch(
+    paddle.reader.shuffle(
+        paddle.dataset.uci_housing.train(), buf_size=500),
+    batch_size=BATCH_SIZE)
 
-```python
-x = paddle.layer.data(name='x', type=paddle.data_type.dense_vector(13))
-y_predict = paddle.layer.fc(input=x,
-                                size=1,
-                                act=paddle.activation.Linear())
-y = paddle.layer.data(name='y', type=paddle.data_type.dense_vector(1))
-cost = paddle.layer.square_error_cost(input=y_predict, label=y)
+test_reader = paddle.batch(
+    paddle.reader.shuffle(
+        paddle.dataset.uci_housing.test(), buf_size=500),
+    batch_size=BATCH_SIZE)
 ```
 
-### Save Topology
+### Train Program Configuration
+The train_program must return the avg_loss as its first returned parameter and then use the inference_program to setup the train_program
```

Comment:
I feel "use the inference_program to setup the train_program" little bit confusing because we did not show inference_program until later